nanoFramework.CoreLibrary/System/Reflection/CustomAttributesHelpers.cs

```diff
@@ -3,74 +3,102 @@
 // See LICENSE file in the project root for full license information.
 //
 
+using System.Collections;
+
 namespace System.Reflection
 {
     internal class CustomAttributesHelpers
     {
-        public static object[] GetCustomAttributesInternal(object[] rawAttributes)
+        /// <summary>
+        /// Creates an array of <see cref="Attribute"/> from a set of attribute definitions.
+        /// </summary>
+        /// <param name="attributeDefinitions">An array of attribute definitions returned from a call to GetCustomAttributesNative.</param>
+        /// <returns>An array of <see cref="Attribute"/>s</returns>
+        /// <remarks>
+        /// <para>
+        /// The attribute definitions are "encoded" into an object array with two positions for each attribute.
+        /// </para>
+        /// 
+        /// <para>
+        /// 1st position - The type of the <see cref="Attribute"/><br/>
+        /// 2nd position - The constructor parameter(s) or null
+        /// </para>
+        ///
+        /// <para>
+        /// Current limitations:<br/>
+        /// </para>
+        /// 
+        /// <para>
+        /// - The parameter(s) have to be a string or numeric type
+        /// - Properties and fields are not supported
+        /// </para>
+        /// </remarks>
+        public static object[] GetCustomAttributesInternal(object[] attributeDefinitions)
         {
-            // get the custom attributes data for the field
-            // these are returned "encoded" in an object array with 2 positions for each attribute
-            // 1st the attribute type
-            // 2nd the constructor parameter or null, if the attribute has no constructor
-            // 
-            // current limitations: 
-            // - works only for constructors with a single parameter
-            // - the parameter has to be a string or numeric type
-            //
-            // both limitations above can be relatively easily overcome by adding the appropriate code at the native handler
+            if (attributeDefinitions is null)
+            {
+                return new object[0];
+            }
 
-            object[] attributes = new object[rawAttributes.Length / 2];
+            var attributes = new ArrayList();
```

Comment:
To improve readability can we have the types instead of `var` on all these please?